newrelic-agent/src/test/java/com/newrelic/agent/config/AgentConfigImplTest.java

```diff
@@ -1293,6 +1296,49 @@ public void getApplicationLoggingConfigSystemProperty() {
                 config.getTransactionTracerConfig().isEnabled());
     }
 
+    @Test
+    public void logFilePriorityOrder() {
```

Comment:
I sequenced the commits so that I verified that this test passed on `main` before layering the remaining changes on it.